File: examples/python/quickstart.py
"""
EdgeBox Python Quickstart - Using FastMCP Client

This example demonstrates how to connect to EdgeBox's MCP server
and use its sandbox capabilities (code execution, file operations,
shell commands, and desktop control) from Python.

Prerequisites:
  1. EdgeBox is running with Docker Desktop
  2. MCP server is healthy (check the EdgeBox dashboard)
  3. Install dependencies: pip install -r requirements.txt

Usage:
  python quickstart.py
"""

import asyncio
from fastmcp import Client


EDGEBOX_MCP_URL = "http://localhost:8888/mcp"


async def main():
    client = Client(EDGEBOX_MCP_URL)

    async with client:
        # --------------------------------------------------------
        # 1. List available tools
        # --------------------------------------------------------
        print("=" * 60)
        print("1. Listing available MCP tools")
        print("=" * 60)
        tools = await client.list_tools()
        for tool in tools:
            print(f"  - {tool.name}: {tool.description}")

        # --------------------------------------------------------
        # 2. Execute Python code
        # --------------------------------------------------------
        print("\n" + "=" * 60)
        print("2. Executing Python code")
        print("=" * 60)
        result = await client.call_tool(
            "execute_python",
            {"code": "import sys; print(f'Hello from EdgeBox! Python {sys.version}')"},
        )
        print(f"  Result: {result}")

        # --------------------------------------------------------
        # 3. Execute Bash commands
        # --------------------------------------------------------
        print("\n" + "=" * 60)
        print("3. Running a shell command")
        print("=" * 60)
        result = await client.call_tool(
            "shell_run",
            {"command": "uname -a && echo '---' && whoami"},
        )
        print(f"  Result: {result}")

        # --------------------------------------------------------
        # 4. File operations - write, read, list
        # --------------------------------------------------------
        print("\n" + "=" * 60)
        print("4. File operations")
        print("=" * 60)

        # Write a file
        await client.call_tool(
            "fs_write",
            {"path": "/tmp/hello.txt", "content": "Hello from EdgeBox Python client!"},
        )
        print("  Written /tmp/hello.txt")

        # Read the file back
        result = await client.call_tool("fs_read", {"path": "/tmp/hello.txt"})
        print(f"  Read back: {result}")

        # List directory
        result = await client.call_tool("fs_list", {"path": "/tmp"})
        print(f"  /tmp contents: {result}")

        # --------------------------------------------------------
        # 5. Execute TypeScript code
        # --------------------------------------------------------
        print("\n" + "=" * 60)
        print("5. Executing TypeScript code")
        print("=" * 60)
        result = await client.call_tool(
            "execute_typescript",
            {"code": "console.log(`Node.js version: ${process.version}`)"},
        )
        print(f"  Result: {result}")

        # --------------------------------------------------------
        # 6. Multi-step workflow: data analysis pipeline
        # --------------------------------------------------------
        print("\n" + "=" * 60)
        print("6. Multi-step workflow: data analysis pipeline")
        print("=" * 60)
        pipeline_code = """
import json

data = [
    {"name": "Alice", "score": 85},
    {"name": "Bob", "score": 92},
    {"name": "Charlie", "score": 78},
    {"name": "Diana", "score": 95},
    {"name": "Eve", "score": 88},
]

avg = sum(d["score"] for d in data) / len(data)
top = max(data, key=lambda d: d["score"])
bottom = min(data, key=lambda d: d["score"])

result = {
    "average_score": avg,
    "top_performer": top["name"],
    "lowest_performer": bottom["name"],
    "total_students": len(data),
}
print(json.dumps(result, indent=2))
"""
        result = await client.call_tool("execute_python", {"code": pipeline_code})
        print(f"  Result: {result}")

        print("\n" + "=" * 60)
        print("All examples completed!")
        print("=" * 60)


if __name__ == "__main__":
    asyncio.run(main())

File: examples/typescript/quickstart.ts
/**
 * EdgeBox TypeScript Quickstart
 *
 * Demonstrates how to connect to EdgeBox's MCP server and use its
 * sandbox capabilities (code execution, file operations, shell commands)
 * from TypeScript using the MCP SDK client (as referenced by fastmcp).
 *
 * Prerequisites:
 *   1. EdgeBox is running with Docker Desktop
 *   2. MCP server is healthy (check the EdgeBox dashboard)
 *   3. Install dependencies: npm install
 *
 * Usage:
 *   npx tsx quickstart.ts
 */

import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { StreamableHTTPClientTransport } from "@modelcontextprotocol/sdk/client/streamableHttp.js";
import { SSEClientTransport } from "@modelcontextprotocol/sdk/client/sse.js";

const EDGEBOX_MCP_URL = "http://localhost:8888/mcp";

async function createClient(): Promise<Client> {
  const client = new Client(
    { name: "edgebox-ts-quickstart", version: "1.0.0" },
    { capabilities: {} }
  );

  // Try Streamable HTTP first, fall back to SSE for compatibility
  try {
    const transport = new StreamableHTTPClientTransport(
      new URL(EDGEBOX_MCP_URL)
    );
    await client.connect(transport);
  } catch {
    console.log("Streamable HTTP not available, falling back to SSE...");
    const sseTransport = new SSEClientTransport(
      new URL("http://localhost:8888/sse")
    );
    await client.connect(sseTransport);
  }

  return client;
}

function printSection(num: number, title: string) {
  console.log(`\n${"=".repeat(60)}`);
  console.log(`${num}. ${title}`);
  console.log("=".repeat(60));
}

async function main() {
  const client = await createClient();

  try {
    // --------------------------------------------------------
    // 1. List available tools
    // --------------------------------------------------------
    printSection(1, "Listing available MCP tools");
    const { tools } = await client.listTools();
    for (const tool of tools) {
      console.log(`  - ${tool.name}: ${tool.description}`);
    }

    // --------------------------------------------------------
    // 2. Execute Python code
    // --------------------------------------------------------
    printSection(2, "Executing Python code");
    const pythonResult = await client.callTool({
      name: "execute_python",
      arguments: {
        code: "import sys; print(f'Hello from EdgeBox! Python {sys.version}')",
      },
    });
    console.log("  Result:", pythonResult.content);

    // --------------------------------------------------------
    // 3. Run a shell command
    // --------------------------------------------------------
    printSection(3, "Running a shell command");
    const shellResult = await client.callTool({
      name: "shell_run",
      arguments: { command: "uname -a && echo '---' && whoami" },
    });
    console.log("  Result:", shellResult.content);

    // --------------------------------------------------------
    // 4. File operations - write, read, list
    // --------------------------------------------------------
    printSection(4, "File operations");

    // Write a file
    await client.callTool({
      name: "fs_write",
      arguments: {
        path: "/tmp/hello.txt",
        content: "Hello from EdgeBox TypeScript client!",
      },
    });
    console.log("  Written /tmp/hello.txt");

    // Read the file back
    const readResult = await client.callTool({
      name: "fs_read",
      arguments: { path: "/tmp/hello.txt" },
    });
    console.log("  Read back:", readResult.content);

    // List directory
    const listResult = await client.callTool({
      name: "fs_list",
      arguments: { path: "/tmp" },
    });
    console.log("  /tmp contents:", listResult.content);

    // --------------------------------------------------------
    // 5. Execute TypeScript code in the sandbox
    // --------------------------------------------------------
    printSection(5, "Executing TypeScript code in sandbox");
    const tsResult = await client.callTool({
      name: "execute_typescript",
      arguments: {
        code: "console.log(`Node.js version: ${process.version}`)",
      },
    });
    console.log("  Result:", tsResult.content);

    // --------------------------------------------------------
    // 6. Multi-step workflow: data analysis pipeline
    // --------------------------------------------------------
    printSection(6, "Multi-step workflow: data analysis pipeline");
    const pipelineCode = `
import json

data = [
    {"name": "Alice", "score": 85},
    {"name": "Bob", "score": 92},
    {"name": "Charlie", "score": 78},
    {"name": "Diana", "score": 95},
    {"name": "Eve", "score": 88},
]

avg = sum(d["score"] for d in data) / len(data)
top = max(data, key=lambda d: d["score"])
bottom = min(data, key=lambda d: d["score"])

result = {
    "average_score": avg,
    "top_performer": top["name"],
    "lowest_performer": bottom["name"],
    "total_students": len(data),
}
print(json.dumps(result, indent=2))
`;
    const pipelineResult = await client.callTool({
      name: "execute_python",
      arguments: { code: pipelineCode },
    });
    console.log("  Result:", pipelineResult.content);

    console.log(`\n${"=".repeat(60)}`);
    console.log("All examples completed!");
    console.log("=".repeat(60));
  } finally {
    await client.close();
  }
}

main().catch(console.error);
